src/Aspire.Cli/Commands/UpdateCommand.cs

```diff
@@ -63,4 +126,257 @@ protected override async Task<int> ExecuteAsync(ParseResult parseResult, Cancell
+                _logger.LogDebug("Creating backup: {BackupPath}", backupPath);
+
+                // Remove old backup if it exists
+                if (File.Exists(backupPath))
+                {
+                    File.Delete(backupPath);
+                }
+
+                // Rename current executable to .old
+                File.Move(targetExePath, backupPath);
+            }
+
+            try
+            {
+                // Copy new executable to install location
+                InteractionService.DisplayMessage("wrench", $"Installing new CLI to {installDir}...");
+                File.Copy(newExePath, targetExePath, overwrite: true);
+
+                // On Unix systems, ensure the executable bit is set
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    SetExecutablePermission(targetExePath);
+                }
+
+                // Test the new executable and display its version
+                _logger.LogDebug("Testing new CLI executable and displaying version");
+                var newVersion = await GetNewVersionAsync(targetExePath, cancellationToken);
+                if (newVersion is null)
+                {
+                    throw new InvalidOperationException("New CLI executable failed verification test.");
+                }
+
+                // If we get here, the update was successful, remove the backup
+                if (File.Exists(backupPath))
+                {
+                    _logger.LogDebug("Update successful, removing backup");
+                    File.Delete(backupPath);
+                }
+
+                // Display helpful message about PATH
+                if (!IsInPath(installDir))
+                {
+                    InteractionService.DisplayMessage("information", $"Note: {installDir} is not in your PATH. Add it to use the updated CLI globally.");
+                }
+            }
+            catch
+            {
+                // If anything goes wrong, restore the backup
+                _logger.LogWarning("Update failed, restoring backup");
+                if (File.Exists(backupPath))
+                {
+                    if (File.Exists(targetExePath))
+                    {
+                        File.Delete(targetExePath);
+                    }
+                    File.Move(backupPath, targetExePath);
+                }
+                throw;
+            }
+        }
+        finally
+        {
+            // Clean up temp directories
+            CleanupDirectory(tempExtractDir);
+            CleanupDirectory(Path.GetDirectoryName(archivePath)!);
+        }
+    }
+
+    private static bool IsInPath(string directory)
+    {
+        var pathEnv = Environment.GetEnvironmentVariable("PATH");
+        if (string.IsNullOrEmpty(pathEnv))
+        {
+            return false;
+        }
+
+        var pathSeparator = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ';' : ':';
```

Comment:
Use Path.PathSeparator.

```suggestion
        var pathSeparator = Path.PathSeparator;
```